### ISSUE TITLE
Tasks with impossible creation dates and streaks

### ISSUE THREAD

**Issue by ghost** [HabitRPG/habitica#4735]
A few days after accepted the [Duolingo learning](https://habitrpg.com/#/options/groups/challenges/2ecaa4b1-529c-415a-8f5a-27223cc93164) challenge from [The Scholars](https://habitrpg.com/#/options/groups/guilds/82fe50b1-4fa5-4e94-8114-aa66516c0d9d) I noticed that the daily tasks associated with it had an impossibly long streak -- as though I'd completed it everyday since creating my account despite not having the challenge. The free rice task is from [another challenge](https://habitrpg.com/#/options/groups/challenges/4d87b94e-e717-45ab-b63b-e486951eb0c0) I accepted at the same time as the Duolingo one, both have been completed everyday. I've not manually adjusted the streak in case that would interfere with finding the bug.

![Task list view](https://cloud.githubusercontent.com/assets/2931465/6318286/f0987ade-ba81-11e4-80de-bfddb047e380.png)

On inspection in the data display tool I noticed that all the tasks associated with the challenge have creation dates in 2014 despite me not joining Habit until Feb 2015! The creation dates for tasks from the free rice challenge were correct.

![Data display after noticing bug](https://cloud.githubusercontent.com/assets/2931465/6318289/0ab44632-ba82-11e4-8298-efbc4c53e8bf.png)

Yesterday I left the challenge but kept the tasks. I've deleted all but the daily and have made edits to the daily; the creation date is still incorrect.

![Data display after leaving challenge](https://cloud.githubusercontent.com/assets/2931465/6318288/ff2b72c2-ba81-11e4-88ea-3070bd4476b2.png)
##

**Comment by ghost** [comment]
Forgot to leave my uuid: 40891154-6548-4dc0-a752-adc284396af0

**Comment by crookedneighbor** [comment]
@ghswa I'm pretty sure those creation dates refer to the date the tasks were created in the challenge, not the date you added them to your profile. I looked at the object for the duolingo challenge and the creation dates listed there match the creation dates on the tasks you have.

**Comment by ghost** [comment]
I still have an impossible streak though. I currently have a 7 day streak which is longer than I've had the task for and longer than the streak shown in the data display.

![screen shot 2015-02-22 at 19 46 53](https://cloud.githubusercontent.com/assets/2931465/6320035/9b5b1b22-bacb-11e4-9a04-72186a531b8f.png)

![Streak history](https://cloud.githubusercontent.com/assets/2931465/6320016/5d546c84-bacb-11e4-9230-7590344d3273.png)

Although I don't doubt you, behaviour wrt creation dates seems to vary from one challenge to another. I've just joined a few extra challenges to double check:
- [Spread the Word Challenge 2015](https://habitrpg.com/#/options/groups/challenges/e1ad6d92-587d-4137-848e-4a1f643603ba): Created on 15-01-01
- [Back-To-School Advice Challenge: WINNERS CHOSEN!](https://habitrpg.com/#/options/groups/challenges/a367eb40-8514-46fd-805e-b9b7f89bad7f): Created on 14-09-12
- [Freerice Challenge](https://habitrpg.com/#/options/groups/challenges/b42e9375-07d8-4996-a191-6d28767db943): Created on 15-02-22
- [15 Minutes on Free Rice Everyday](https://habitrpg.com/#/options/groups/challenges/4d87b94e-e717-45ab-b63b-e486951eb0c0): Created on 15-02-22

That last challenge definitely wasn't created today and I'm pretty sure the third wasn't either.

**Comment by CindyAH** [comment]
I also have this issue. I started HabitRPG on 4/1/2015, but today (April 19) noticed that I have a 21-day streak.  I would have created these tasks on April 1 or 2, and my first stats & streaks backup is April 6.  On that day, for "brush teeth", I already have a 7, and for exercise, I already have a 6.  However, for both of them, I would not have even checked it off, since I do that in the evening for both those tasks.  So I have a free day for each, and possibly two free days.  I have not fixed my streaks.

UUID <b>cae108a4-5ba4-400b-a37f-39df47376b0b</b>
    From my email to me:
<b>Apr 6 (13 days ago)</b>

to me 
Stats and Streaks Backup

...

Your Dailies' Streaks

0 Troll for new members
1 30 minutes yard work
2 Read Art Of Unit Testing until finished for the week
3 out of bed within 5 minutes of alarm
3 dressed with food before going to computer
5 Leave by 7:20
3 Take stairs
3 play ping pong (or invite two people and fail)
3 Stand at desk for two hours
2 Update TFS
<b>6 square dance or ride bike (20) or walk (30)
7 Brush teeth</b>

**Comment by deilann** [comment]
Can you please post your UUID, which can be found in settings > API?
Without it, we can't do any investigation.

**Comment by CindyAH** [comment]
Yeah, I edited my post almost immediately - sorry!  Feel free to ask for
anything else as well.

**Comment by deilann** [comment]
Most of the people who will be able to help you receive these tickets via
email notifications. If you edit a post, it does not send a new email
notification. It is almost always better to add a new message, rather than
edit. If you feel that makes things cluttered, you can delete the message
afterwards and edit it into the old one.

**Comment by deilann** [comment]
Did you ever manually mess with the streaks on your tasks? (You can do this
if you open up the tasks to edit.)

If you'd like to get rid of the streak you did not earn, you can do this by
going to Site > Settings > Fix Character Values and set your number of
Streaks to 0.

**Comment by deilann** [comment]
Do you ever have Habit open in multiple tabs on your browser?

**Comment by CindyAH** [comment]
I also have this issue. I started HabitRPG on 4/1/2015, but today (April 19) noticed that I have a 21-day streak. I would have created these tasks on April 1 or 2, and my first stats & streaks backup is April 6. On that day, for "brush teeth", I already have a 7, and for exercise, I already have a 6. However, for both of them, I would not have even checked it off, since I do that in the evening for both those tasks. So I have a free day for each, and possibly two free days. I have never messed with my streaks manually, or done anything else in the "fix character values" section.  <b>I often have it open in multiple tabs in my browser.</b>

<b>UUID cae108a4-5ba4-400b-a37f-39df47376b0b</b>
<b>From my email to me:
Apr 6 (13 days ago)</b>

to me 
Stats and Streaks Backup

...

Your Dailies' Streaks

0 Troll for new members
1 30 minutes yard work
2 Read Art Of Unit Testing until finished for the week
3 out of bed within 5 minutes of alarm
3 dressed with food before going to computer
5 Leave by 7:20
3 Take stairs
3 play ping pong (or invite two people and fail)
3 Stand at desk for two hours
2 Update TFS
<b>6 square dance or ride bike (20) or walk (30)
7 Brush teeth</b>

**Comment by deilann** [comment]
Yeah, the issue is that you have it open in multiple tabs of your browser.

Here's what's happening:

You complete the Daily. You go to a tab where you haven't completed that
Daily and you check it off again. Now you've completed it twice in one day,
doubling your streak.

How to prevent this:

Don't have Habit open in multiple tabs.

-or-

Sync before checking things off.

**Comment by CindyAH** [comment]
Cool, thanks.  Don't know if I can have ANYTHING open in a single tab, but I'll try!  :)

**Comment by deilann** [comment]
Well, if you sync before checking off tasks or try to reliably use the same
tab, things should be okay!

**Comment by dhouck** [comment]
I also experienced the same problem yesterday, and I _am_ using only one tab. I created a few dailies yesterday, and then by the end of the day had streak bonuses of 4 for some of them. I was having other syncing issues at the time, too, though; for example, the data display tool wasn’t updating even after I hit sync on the HabitRPG tab and re-fetched the data on the tool.

I’ve since fixed the streaks to the correct values, but my user ID GUID is 9d72049b-e358-4658-af5b-0957afdb3f10.
